app/src/main/java/org/mozilla/focus/whatsnew/WhatsNew.kt

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.whatsnew
+
+import android.content.Context
+import android.preference.PreferenceManager
+import android.support.annotation.VisibleForTesting
+import org.mozilla.focus.ext.appVersionName
+
+class WhatsNew {
+    companion object {
+        // TODO: Update to final URL
+        const val SUMO_URL = "https://support.mozilla.org/en-US/products/focus-firefox/firefox-focus-android"
+
+        @VisibleForTesting const val PREFERENCE_KEY_APP_NAME = "lastKnownAppVersionName"
+        private const val PREFERENCE_KEY_SESSION_COUNTER = "updateSessionCounter"
+        private const val SESSIONS_PER_UPDATE = 3
+
+        @VisibleForTesting var wasUpdatedRecently: Boolean? = null
+
+        fun wasUpdatedRecently(context: Context): Boolean {
+            if (wasUpdatedRecently == null) {
+                wasUpdatedRecently = when {
+                    hasBeenUpdated(context) -> {
+                        rememberAppVersionName(context)
+                        setSessionCounter(context)
+                        true
+                    }
+                    else -> hasBeenUpdatedInRecentSession(context)
+                }
+            }
+            return wasUpdatedRecently!!
```

Comment:
nit:  The reason smart cast to a non-null type doesn't work here is because `wasUpdatedRecently` is mutable so another thread could modify it to be null while this function is running. You can make smart cast work by storing this value in a constant `val` at the top of the function and using that in the function instead. Personally, I prefer this over `!!` because you're less likely to run into a crash if you ever change the file (e.g. this return gets accidentally moved inside the if).